src/ansys/mapdl/core/mapdl.py

```diff
@@ -2777,7 +2777,10 @@ def run(self, command, write_to_log=True, mute=None, **kwargs) -> str:
         if "\n" in command or "\r" in command:
             raise ValueError("Use ``input_strings`` for multi-line commands")
 
-        if self._store_commands:
+        # check if we want to avoid the current non-interactive context.
+        avoid_non_interactive = kwargs.get("avoid_non_interactive", False)
```

Comment:
```suggestion
        avoid_non_interactive = kwargs.pop("avoid_non_interactive", False)
```
Because the argument is not needed in ``kwargs`` later, we can do ``pop`` so it is removed too.

Reply:
Pinging @clatapie @maromer0 for learning purposes